src/core/ddsc/tests/deadline.c

```diff
@@ -434,3 +437,117 @@ CU_Theory((int32_t n_inst, uint8_t unreg_nth, uint8_t dispose_nth), ddsc_deadlin
     }
   } while (!test_finished);
 }
+
+#define DEADLINE DDS_MSECS(500)
+
+static void cb (struct ddsi_xevent *xev, void *ptr, ddsrt_mtime_t tm)
+{
+  (void) xev;
+  (void) ptr;
+  (void) tm;
+  //this function's purpose is to delay the monitor thread such that while instance's deadline may expire, but their updates are "queued" if updates happen while the monitor thread is "busy"
+  dds_sleepfor(DEADLINE);
+}
+
+static void check_statuses(dds_entity_t datareader, dds_entity_t datawriter, uint32_t prevcount, uint32_t newcount, dds_instance_handle_t prevhandle, dds_instance_handle_t newhandle)
+{
+  dds_requested_deadline_missed_status_t rstatus;
+
+  dds_return_t rc = dds_get_requested_deadline_missed_status (datareader, &rstatus);
+  CU_ASSERT_EQUAL_FATAL(rc, DDS_RETCODE_OK);
+  CU_ASSERT ( (rstatus.total_count == prevcount && rstatus.last_instance_handle == prevhandle) ||
+              (rstatus.total_count == newcount  && rstatus.last_instance_handle == newhandle));
+
+  dds_offered_deadline_missed_status_t ostatus;
+
+  rc = dds_get_offered_deadline_missed_status (datawriter, &ostatus);
+  CU_ASSERT_EQUAL_FATAL(rc, DDS_RETCODE_OK);
+  CU_ASSERT ( (ostatus.total_count == prevcount && ostatus.last_instance_handle == prevhandle) ||
+              (ostatus.total_count == newcount  && ostatus.last_instance_handle == newhandle));
+}
+
+CU_Test(ddsc_deadline, update)
+{
+  dds_entity_t pp = dds_create_participant(DDS_DOMAIN_DEFAULT, NULL, NULL);
+  CU_ASSERT_FATAL(pp > 0);
+
+  dds_entity_t tp = dds_create_topic(pp, &Space_Type1_desc, "abc_def", NULL, NULL);
```

Comment:
Bummer, I overlooked this one the first time round ...

```suggestion
  char topicname[100];
  create_unique_topic_name ("ddsc_deadline_update", topicname, sizeof topicname);
  dds_entity_t tp = dds_create_topic(pp, &Space_Type1_desc, topicname, NULL, NULL);
```

Reply:
Ouch thanks for catching that one!

Seems like I was way too focussed on the correct sequencing of reads/writes/sleeps/etc. and not giving enough attention to this side of the testing business.